packages/plugins/typescript/resolvers/src/index.ts

```diff
@@ -71,6 +71,22 @@ export type NewStitchingResolver<TResult, TParent, TContext, TArgs> = {
       context: TContext,
       info: GraphQLResolveInfo
     ) => Promise<TResult> | TResult;`);
+
+    defsToInclude.push(`export type RecursivePick<T, S> = Pick<
+      {
+        [K in keyof T & keyof S]: 
+          S[K] extends true ? 
+            T[K]
+            : Maybe<T[K]> extends T[K] ?
+              NonNullable<T[K]> extends (infer U)[] ?
+                Maybe<RecursivePick<NonNullable<U>, S[K]>>[]
+                : Maybe<RecursivePick<NonNullable<T[K]>, S[K]>>
+              : T[K] extends (infer U)[] ?
+                RecursivePick<U, S[K]>[]
+                : RecursivePick<T[K], S[K]>;
+      },
+      keyof T & keyof S
+    >;`);
```

Comment:
Maybe we could split it into 2-3 types to make it more readable?

Reply:
Good suggestion. I split up the types to be a bit more readable. Can you take another look?